guardian/pressreader
# Replace ts-node-dev with tsx

## What does this change?

Replace `ts-node-dev` with `tsx` for use as the dev 'server' for the app.

The initial impetus for looking at this was because `ts-node-dev` introduces a vulnerability via a transitive dep (though the fact that it's in the dev server package seems to limit the scope considerably). But when investigating this I saw that the package is largely unmaintained (last release was 2 years ago). `ts-node-dev` is a wrapper around `ts-node`, so `ts-node` could have made sense as a replacement. However, `tsx` seems to be favoured in the department moving forward for, and has a build in 'watch' mode, which is all we need.



## How to test

Install new npm dependencies, run `docker-compose up` from the root of the repo, and then `npm run dev -w pressreader`. The script should run (nb. it's always been a bit dodgy locally when it comes to making the HTTP requests to CAPI for some reason) and should re-run when you save changes to `handler.ts`.



## How can we measure success?



## Have we considered potential risks?



## Images



## Accessibility
